pahfit/base.py

```diff
@@ -0,0 +1,462 @@
+        nan designates no limit
+
+    Returns
+    -------
+    plimits : list of tuples
+        tuples give the min/max limits for a parameter
+    """
+    plimits = []
+    for cmin, cmax in zip(min_vals, max_vals):
+        if np.isnan(cmin):
+            cmin = None
+        if np.isnan(cmax):
+            cmax = None
+        plimits.append((cmin, cmax))
+
+    return plimits
+
+
+def _ingest_fixed(fixed_vals):
+    """
+    Ingest the fixed value read from a file and generate the appropriate
+    internal format (list of booleans).  Needed as booleans are
+    represented in files as True/False strings.
+
+    Parameters
+    ----------
+    min_vals : numpy.array (string)
+        fixed designations
+
+    Returns
+    -------
+    pfixed : numpy.array (boolean)
+        True/False designation for parameters
+    """
+    pfixed = []
+    for cfixed in fixed_vals:
+        if cfixed == 'True':
+            cfixed = True
+        if cfixed == 'False':
+            cfixed = False
+        pfixed.append(cfixed)
+
+    return pfixed
+
+
+class PAHFITBase():
+    """
+    Base class for PAHFIT variants. Each variant nominally specifies the valid
+    wavelength range, instrument, and type of astronomical objects.
+
+    For example, the original IDL version of PAHFIT was valid for
+    Spitzer/IRS spectra (5-38 micron) and observations of parts or all of
+    external galaxies.
+
+    Mainly sets up the astropy.modeling compound model
+    that includes all the different components including
+    blackbodies for the continuum, drudes for the dust
+    emission features, and Gaussians for the gas emission features.
+
+    Parameters
+    ----------
+    filename: string
+        filename giving the pack that contains all the
+        info described for param_info
+
+    tformat: string
+        table format of filename (compatible with astropy Table.read)
+
+    param_info: tuple of dics
+        The dictonaries contain info for each type of component.  Each
+        component of the dictonaries is a vector.
+        bb_info -
+        dict with {name, temps, temps_limits, temps_fixed,
+        amps, amps_limits, amps_fixed}, each a vector
+        dust_features, h2_features, ion_features -
+        dict with {name amps, amps_limits, amps_fixed,
+        x_0, x_0_limits, x_0_fixed, fwhms, fwhms_limits, fwhm_fixed}.
+    """
+
+    def __init__(self,
+                 param_info=None,
+                 filename=None,
+                 tformat=None):
+        """
+        Setup a variant based on inputs.  Generates an astropy.modeling
+        compound model.
+
+        Notes
+        -----
+        Would be great to rename the parameters such that they uniquely
+        identify the component (dust, gas, specific line, etc.).  This is
+        possible - say a discussion on the stsci slack channel - James Davies?
+        """
+        # check that param_info or filename is set
+        if filename is None and param_info is None:
```

Comment:
Should it be: if filename is None or param_info is None:

Reply:
I guess param_info contains saved results that are loaded back(?), so in that case "and" is correct.

Reply:
Nope.  As long as one is not None, then things are ok.  Basically, either the filename or the param_info needs to be passed.  If neither are, then this is an error.